### PR TITLE
Towards testing with cabal

### PR DESCRIPTION
**Pull request by DanielG**
Currently the tests force using `stack`, but HIE should be an equal opportunity build system employer :)

Remember to read commit messages for details.

---

TODO:
- [ ] Also test against cabal cradle

### CODE REVIEW

File: test/utils/TestUtils.hs [hunk 1]
```diff
@@ -159,9 +159,6 @@ stackYaml =
   "stack-8.0.2.yaml"
 #endif
 
-logFilePath :: String
-logFilePath = "functional-hie-" ++ stackYaml ++ ".log"
```
Comment (power-fungus) [inline test/utils/TestUtils.hs]: This function is required by some tests. See ci-build

Comment (DanielG) [reply test/utils/TestUtils.hs]: A grep over the entire source doesn't show any other users of this definition. I don't know what you mean?

Comment (DanielG) [reply test/utils/TestUtils.hs]: Ugh, nvm. I'm an idiot :)

**Comment by power-fungus** [comment]
I actually tried to run the tests with cabal and a large number of them failed. See [my azure-ci build](https://dev.azure.com/haskell-ide-engine/haskell-ide-engine-power-fungus/_build/results?buildId=140) for a comparison between the test-runs using `cabal` and `stack`.

I would say that currently executing the tests with stack is unsupported as well. @jneira, @fendor and I tried our best in the last months to get the tests working using a repeatable script that starts from a clean os-image. We did not succeed so far.

**Comment by DanielG** [comment]
I ran all the tests with cabal on my machine and they all worked perfectly after all the necessary `hoogle` setup and what have you at least. But I agree the tests are very fiddly at the moment.

**Comment by DanielG** [comment]
Looks like your CI builds with cabal were all failing because of caba-helper $libexec stuff. Not sure why that was working for me..

**Comment by power-fungus** [comment]
The tests work in specialized environments that nobody knows how to create. The azure-ci scripts help to find a way to create such an environment reliably.

Do the tests still work on your machine after cleaning `~/.cabal`, running `git clean -Xdf` in the hie-repo and removing all `hie` and `cabal-helper` executables, unsetting ?

**Comment by DanielG** [comment]
Indeed it looks like I had an hie executable in `~/.local/bin` from running `stack install` still. I don't understand why it makes a difference as new-test does put the cabal installed hie executable on PATH before `~/.local` though. Strange.

**Comment by power-fungus** [comment]
> Strange.

This describes the `hie` runtime-dependencies perfectly well. To this date, I still have no idea what is really needed and why `hie` fails on certain projects or setups. The azure-ci and `install.hs` projects were efforts towards a deeper understanding of the issue. I have learned some rules, but I am no less confused.

The [curent build](https://dev.azure.com/haskell-ide-engine/haskell-ide-engine-power-fungus/_build/results?buildId=140) shows that the tests also fail with `stack`. Some environment is no quite right. :cry:

**Comment by fendor** [comment]
So, this looks pretty nice now, right? I would be so glad if we could remove stack as a test dependency :)

**Comment by jneira** [comment]
@fendor If i have ubderstood correctly tests started to fail in the last @DanielG, right?

**Comment by DanielG** [comment]
The tests did fail once they were using the `hie` executable built by cabal. I still think we should merge this. It's at least making some progress towards getting cabal working.

I was going to include more stuff here but got side-tracked a bit by cabal-helper work so I think we should just merge this now.

**Pull request merged**